src/receipts.ts

```diff
@@ -0,0 +1,241 @@
+};
+
+type receiptDispatcherMap = {
+  [id: string]: receiptEmailStatus;
+};
+
+@customElement('iaux-mgc-receipts')
+export class IauxMgcReceipts extends LitElement {
+  @property({ type: Array }) receipts = [];
+
+  @property({ type: Object })
+  receiptDispatcher: receiptDispatcherMap | null = null;
+
+  shouldUpdate(changed: PropertyValues) {
+    if (changed.has('receiptDispatcher')) {
+      return true;
+    }
+    return false;
+  }
+
+  updated(changed: PropertyValues) {
+    if (changed.has('receipts')) {
+      this.updateReceiptSentMap();
+    }
+    if (changed.has('receiptDispatcher')) {
+      console.log('receiptDispatcher UPDATED ---- ', this.receiptDispatcher);
+    }
+  }
+
+  updateReceiptSentMap() {
+    if (!this.receipts.length) {
+      this.receiptDispatcher = null;
+    } else {
+      const receiptDispatcher: receiptDispatcherMap = {};
+      this.receipts.forEach((receipt: Receipt) => {
+        receiptDispatcher[receipt.id] = {
+          id: receipt.id,
+          emailStatus: '',
+        };
+      });
+      this.receiptDispatcher = receiptDispatcher;
+    }
+  }
+
+  emailReceipt(receipt: Receipt) {
+    this.dispatchEvent(
+      new CustomEvent('EmailReceiptRequest', {
+        detail: {
+          donation: receipt,
+        },
+      })
+    );
+  }
+
+  donationAmountFormatted(amount: number) {
+    return `USD $${amount}`;
+  }
+
+  /** callback that confirms status of an receipt email request  */
+  async emailSent(receiptEmailed: receiptEmailStatus) {
+    const currStatusMap = this.receiptDispatcher;
+    this.receiptDispatcher = null;
+    await this.updateComplete;
+    const statusMap = {
+      ...currStatusMap,
+    } as receiptDispatcherMap;
+    const { id } = receiptEmailed;
+    statusMap[id] = receiptEmailed;
+
+    this.receiptDispatcher = { ...statusMap };
+    console.log(
+      'RECEIPTS -- emailSent',
+      this.receiptDispatcher,
+      receiptEmailed
+    );
+
+    // re-enable email request button
+    const escapedId = CSS.escape(id);
+    const button = this.shadowRoot?.querySelector(
+      `#donation-${escapedId} iaux-button`
+    ) as IauxButton;
+    button.isDisabled = false;
+  }
+
+  /* renderings */
+  emailStatusMessageToDisplay(receiptSentStatus: receiptEmailStatus): string {
+    switch (receiptSentStatus.emailStatus) {
+      case 'success':
+        return '✓ Sent';
+      case 'fail':
+        return '✖ Failed';
+      default:
+        return '';
+    }
+  }
+
+  ctaButtonText(donation: Receipt, emailStatus?: receiptEmailStatus) {
+    if (emailStatus?.emailStatus === 'pending') {
+      return 'Sending...';
+    }
+
+    return 'Email receipt';
+  }
+
+  protected render() {
+    return html`
+      <section id="recent-donations-list">
+        <table>
+          <tr>
+            <th class="date">Date</th>
+            <th class="amount">Amount</th>
+            <th class="action">Action</th>
+          </tr>
+          ${this.receipts.length
```

Comment:
Minor, but to make this `render()` method a bit easier to read, I might split out the receipts list view, ie:
```
${this.receipts.length > 0 : this.receiptsListView : html`<p>No recent donations found</p>`}

func receiptsListView(): TemplateResult {
  return html`...`
}
```